barretenberg/cpp/src/barretenberg/relations/poseidon2_external_relation.hpp

```diff
@@ -0,0 +1,121 @@
+#pragma once
+#include "barretenberg/relations/relation_types.hpp"
+namespace proof_system {
+
+template <typename FF_> class Poseidon2ExternalRelationImpl {
+  public:
+    using FF = FF_;
+
+    static constexpr std::array<size_t, 4> SUBRELATION_PARTIAL_LENGTHS{
+        7, // external poseidon2 round sub-relation for first value
+        7, // external poseidon2 round sub-relation for second value
+        7, // external poseidon2 round sub-relation for third value
+        7, // external poseidon2 round sub-relation for fourth value
+    };
+
+    /**
+     * @brief Expression for the poseidon2 external gate.
```

Comment:
Would you please explain what exactly is being calculated here in terms of the notation of the paper? I say more about that below. I think the intent is the external round function of Section 6 of the paper, but I'm worried there's a discrepancy: looks like you're using $M_{\mathcal{E}} = M_4$, but to my eyes it looks like there might be a missing 2, ie should we have $M_{\mathcal{E}} = 2M_4$? I didn't do a thorough read of the paper--did you look into this?